src/aliases/mapper/Befehle/minit.lua

```diff
@@ -1,16 +1,41 @@
-
 -- Dies setzt die Karte auf Werkszustand zurueck.
 
-for i,name in pairs(getRooms()) do
-    deleteRoom(i)
+-- Sonderfälle, behandeln wir irgendwann in der Zukunft
+if gmcp.MG.room.info == nil or gmcp.MG.room.info.id == nil then
+    echoM("Bitte begebe dich in einen anderen Raum und versuche es noch einmal! Habe derzeit keinerlei Informationen ueber deinen aktuellen Standort im MorgenGrauen :(")
+    return
+end
```

Comment:
Das finde ich sehr hart. Es ist auch schlechter als zuvor. Dabei treffen die meisten folgenden Aktionen doch trotzdem noch zu, auch wenn man in einem unbekannten Raum steht, oder nicht? Einzig der neue Raum bekommt dann weniger Infos. Aber den Rest der Map kann man löschen und den neuen Raum trotzdem anlegen.

edit: In der Praxis scheint das selten vorzukommen. Selbst wenn man in einen leeren Raum geht, hat GMCP noch die Infos vom Raum davor gespeichert. Der merkt schlicht gar nicht, dass man den Raum gewechselt hatte. Ich konnte das hier nur nachstellen, indem ich in einem dunklen Raum eingeloggt habe. Welches Problem hattest du angepeilt?

Reply:
Der Test klappt so nicht.

```lua
gmcp.MG = {} 

-- gmcp.MG.room == nil -> true
-- gmcp.MG.room.info == nil -> false da schon gmcp.MG.room nil ist
```

Hatte es vorhin probiert in einem dunklen Raum zu reloggen und dann minit aufzurufen. Dabei aber auch nicht bedacht das Mudlet bei einem Relog den Inhalt der gmcp Tabelle NICHT leert.

Reply:
Sinn und Zweck des Tests dort ist das man vorerst keine neue Karte beginnen kann wenn man sich in einem Sonderraum befindet. Allerdings stammt er aus einer Version der Skripte die schon mit Finsternis umgehen kann. Das wollte ich aber in einem extra branch machen, weil ich da noch ausgiebig weitertesten muss.

Ich überlege mir was anderes.